[quant][graphmode][fix] Run preprocess for child module before parent module

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39380 [quant][graphmode] Test for another type of ops in insert_observer for if
* **#39368 [quant][graphmode][fix] Run preprocess for child module before parent module**
* #39367 [quant] Fix fusion pattern for add_relu

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21829296](https://our.internmc.facebook.com/intern/diff/D21829296)

jerryzh168 committed:
[quant][graphmode][fix] Run preprocess for child module before parent module

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode][fix] Run preprocess for child module before parent module"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21829296](https://our.internmc.facebook.com/intern/diff/D21829296)

[ghstack-poisoned]

supriyar commented on test/quantization/test_quantize_script.py:
```diff
@@ -695,6 +695,43 @@ def forward(self, x):
+            def forward(self, x):
+                if self.use_skip:
+                    if self.cond:
+                        return self.conv(x)
```
Should we test with different ops in each conditional statement? Can help increase the coverage of this test.

jerryzh168 replied:
Yeah, I think we can test different types of ops here, but maybe we can remove "nested" and just test `if`, since there is actually not much difference between `if` and `nested if`. I'll add one more test for a different type

supriyar reviewed (approved):
lg, left a comment about updating test

jerryzh168 committed:
Update on "[quant][graphmode][fix] Run preprocess for child module before parent module"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21829296](https://our.internmc.facebook.com/intern/diff/D21829296)

[ghstack-poisoned]
